src/services/SolanaService.ts

```diff
@@ -201,14 +202,13 @@ export class SolanaService extends NamingService {
     return records
   }
 
-  async addr (name: string): Promise<RecordItemAddr> {
+  async addr (name: string): Promise<RecordItemAddr | null> {
     const recordItem = this.makeRecordItem(`${KeyPrefix.Address}.sol`)
     const { pubkey } = await getDomainKey(name)
     const isValid = await this.isValidPubkey(pubkey)
-    if (isValid) {
-      const address = await resolve(this.provider, name);
-      recordItem.value = address.toString()
-    }
+    if (!isValid) return null
```

Comment:
Will kind of way misguided the developers? Or is it better to throw a relevant error?